### ISSUE TITLE
Help with issue in tutorial: Coordinate Transforms using TF

### ISSUE THREAD

**Issue by jwhendy** [ros-industrial/industrial_training#137]
Hello,

I'm pretty new to ROS and am using `python`. I've been doing pretty well translating your tutorials (which are awesome, btw), but am hung up on the `tf` one [here](https://github.com/ros-industrial/industrial_training/wiki/Coordinate-Transforms-using-TF).

I went through the [normal ROS tf tutorials](http://wiki.ros.org/tf/Tutorials) and tried to apply what I learned there to translate yours. What's goofing me is this error:

```
[ERROR] [1499188908.643911]: Error processing request: "world" passed to lookupTransform argument target_frame does not exist. 
['Traceback (most recent call last):\n', '  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/impl/tcpros_service.py", line 625, in _handle_request\n    response = convert_return_to_response(self.handler(request), self.response_class)\n', '  File "/home/jwhendy/catkin_ws/src/workcell_core/src/vision_node.py", line 46, in localizePart\n    rospy.Time(0))\n', 'LookupException: "world" passed to lookupTransform argument target_frame does not exist. \n']
```

I tried starting up the launch file to that point (no `tf` stuff included) and then manually looking up the transform from `world` to `camera_frame` as it looks like you're suggesting:

```
$ rosrun tf tf_echo world camera_frame
Failure at 1499189021.884116537
Exception thrown:"world" passed to lookupTransform argument target_frame does not exist. 
The current list of frames is:

Failure at 1499189021.884201418
Exception thrown:"world" passed to lookupTransform argument target_frame does not exist.
```

The code appears to be working otherwise:

```
$ roslaunch workcell_support workcell.launch 
... logging to /home/jwhendy/.ros/log/47d4e9ec-60dc-11e7-b011-cc3d82b9eef7/roslaunch-roboxenial-8664.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://roboxenial:43051/

SUMMARY
========

PARAMETERS
 * /rosdistro: kinetic
 * /rosversion: 1.12.7
 * /workcell_node/base_frame: world

NODES
  /
    fake_ar_publisher (fake_ar_publisher/fake_ar_publisher_node)
    vision_node (workcell_core/vision_node.py)
    workcell_node (workcell_core/workcell_node.py)

auto-starting new master
process[master]: started with pid [8675]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 47d4e9ec-60dc-11e7-b011-cc3d82b9eef7
process[rosout-1]: started with pid [8688]
started core service [/rosout]
process[fake_ar_publisher-2]: started with pid [8695]
process[vision_node-3]: started with pid [8706]
process[workcell_node-4]: started with pid [8718]
[ INFO] [1499188487.648367480]: Starting simulated ARMarker publisher
[INFO] [1499188487.906461]: Vision node up
[INFO] [1499188488.182812]: Scan-N-Plan initialized
[INFO] [1499188488.183807]: requesting pose in: world
[INFO] [1499189503.933117]: pose = position: 
  x: -0.6
  y: 0.2
  z: 0.5
orientation: 
  x: 0.0
  y: 0.0
  z: 0.0
  w: 1.0
```

I guessed this frame is to be provided by `fake_ar_publisher`, but see this in the code:
```
// Given a marker w/ pose data, publish an RViz visualization
// You'll need to add a "Marker" visualizer in RVIZ AND define
// the "camera_frame" TF frame somewhere to see it.
```

Does that mean I need to define `camera_frame` somewhere? I'm a little puzzled by trying to adapt the `tf` tutorials themselves as the `lookupTransform` is done between `/turtle2` and `/turtle1`... I'm assuming these must be frame names created by the turtles,  but they have sort of topic looking names so I'm not really sure.

Could anyone help point me in the right direction?

---

Edit: this is on Ubuntu 16.04 and ROS kinetic.

**Comment by jwhendy** [comment]
Baaaahhhh. I swear it's always after writing an issue that I have some new insight (despite trying at this for a couple of days). I just realized that this is defined in the `workcell.xacro` files in the previous tutorials. If I start both `workcell.launch` and `urdf.launch`  I can translate the two frames with the `tf_echo` command described above.

I completely missed this from the tutorial conclusion :(

```
roslaunch myworkcell_support urdf.launch
roslaunch myworkcell_support workcell.launch
```

I'm still having issues with doing the transform (my `waitForTransform` times out), but at least I can echo the transform, so I think I've ironed out the main issue... Silly me and hopefully this helps someone else.

**Comment by jwhendy** [comment]
As another update to this, I finally solved the overall issue. I don't know if this is `rospy` specific, but you might want to update your tutorials to use `tf2`, not `tf`. Because `camera_frame` is defined as fixed and it seems that fixed frames were updated to be `tf2` types. I found a bunch of answers very similar to [this one](http://answers.ros.org/question/231914/tf_echo-can-find-transform-but-waitfortransform-cant/) on ROS Answers (after I finally figured out what in the world I was looking for!).
